src/iceberg/schema.h

```diff
@@ -29,11 +29,23 @@
 #include <vector>
 
 #include "iceberg/iceberg_export.h"
+#include "iceberg/result.h"
 #include "iceberg/schema_field.h"
 #include "iceberg/type.h"
 
 namespace iceberg {
 
+/// \brief Transparent hash function that supports std::string_view as lookup key
+///
+/// Enables std::unordered_map to directly accept std::string_view lookup keys
+/// without creating temporary std::string objects, using C++20's transparent lookup.
+struct string_hash {
```

Comment:
Do we want to move it to string_utils.h so type.h can reuse it?

Reply:
Perhaps we can do this altogether in https://github.com/apache/iceberg-cpp/pull/194